### ISSUE TITLE
Cannot create a second steam app folder or move games to a separate folder

### ISSUE THREAD

**Issue by good-old-music** [ValveSoftware/steam-for-linux#5737]
#### Your system information
    CPU Vendor:  GenuineIntel
    CPU Brand:        Intel(R) Xeon(R) CPU E3-1240 V2 @ 3.40GHz
    Speed:  3800 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
Operating System Version: Debian Buster
    Linux 4.x (64 bit)
    Kernel Version:  4.17.0-3-amd64
Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 1050/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 390.77
    OpenGL Version: 4.6
Sound card:
    Audio device: Realtek ALC269VB
Memory:
    RAM:  15998 Mb
* Steam client version (build number or date):     Steam Runtime Version:  steam-runtime-beta-release_2018-06-14
* Distribution (e.g. Ubuntu): Debian Buster via Flatpak
* Opted into Steam client beta?: [Automatic]
* Have you checked for system updates?: [Automatic]

#### Please describe your issue in as much detail as possible:
Steam is installed via Flatpak so games run as if under Ubuntu variants

Mounted partition as /app
tried to install games to /app
[create folder - SteamLibrary] Steam refuses as drive is read only, (it isn't) message - [Failed to create a folder, this drive is read only.]
[just install ] Steam refuses as drive is not empty (it is) message - [Failed to add new Steam library folder]

I had originally tried to just move games to an new partition as my drive is beginning to fill up but it refuses to acknowledge their presence

This may be because I run Steam through a Flatpak and it therefore cannot access my system - however I would appreciate any help you can offer to make a new partition available.
#### Steps for reproducing this issue: 

1. Install via Flatpak
2. see above
3.

**Comment by FateRover** [comment]
same DEBIAN BUSTER ,U  should mout the disk before steam start, start steam first and mount disk will lead this.

**Comment by good-old-music** [comment]
Thanks FateRover but, the disk is automounted at boot so a different issue I guess.

**Comment by FateRover** [comment]
suggest close steam ，unmount the disk，remount it， then start steam. if ntfs format ,follow  this https://github.com/ValveSoftware/steam-for-linux/issues/5742#issuecomment-420753840   .maybe help ,but  i dont use  flatpak,just install steam via debian source

**Comment by good-old-music** [comment]
Thank you for your comments FateRover,

Using a Steam client would certainly be the long run option I guess but does involve much reinstallation of games etc.

I believe after some more consideration the issue is down to Flatpak itself, it creates a closed system that doesn't allow for accessing the filesystem outside of the Flatpak installation, this is a security measure.
The real question I suppose is how to put a new filesystem inside the existing Flatpak that the client can use?
This would not I guess be an issue with Steam client although an answer may also be useful to others besides myself.
Maybe the best bet is to expand the existing partition into unused H/D space.
tldr,
---
If I have time enough to mess around a bit, I will investigate further with the Flatpak rather than Steam and hopefully come up with an answer that doesn't involve expanding partitions and reinstalling the Flatpak and associated Steam games too. Hmmm, maybe hard linking a folder to a partition inside the Flatpak would work but it would have to conform somehow to Steams existing, expected folder naming to allow installs, and that wouldn't let me use the ssd as you can't hardlink filesystems on a separate drive.
Debian source hasn't had a Steam client for some while in Buster, and the version for Stretch was essentially broken for installing on later systems due to updated system libraries. I think there is a version in sid which is somewhat experimental and hasn't yet passed the requirements for putting into testing, although again, this would involve re-installing games and mods.

**Comment by BoltsJ** [comment]
You need to add a flatpak override to give steam access to the additional folders. `flatpak override --user --filesystem=/path/to/folder com.valvesoftware.Steam`

**Comment by nanonyme** [comment]
Since this is a quite common thing to want, there has been a FAQ entry https://github.com/flathub/com.valvesoftware.Steam/wiki/Frequently-asked-questions#i-want-to-add-external-disk-for-steam-libraries for a longer time.

**Comment by kisak-valve** [comment]
Closing because there's no action to take by a Valve dev. This is part of the nature of Flatpak.
